lib/Model/TypeSystemPrinter.cpp

```diff
@@ -127,32 +122,42 @@ TypeSystemPrinter::~TypeSystemPrinter() {
   Out.flush();
 }
 
-/// Build a C-like string for a given QualifiedType
-static llvm::SmallString<32>
-buildFieldName(const model::QualifiedType &FieldQT) {
-  llvm::SmallString<32> FieldName;
+/// Build a C-like string for a given Type
+static std::string buildFieldName(const model::Type &Type,
+                                  std::string &&Prefix = {},
+                                  std::string &&Suffix = {}) {
+  if (const auto *Array = llvm::dyn_cast<model::ArrayType>(&Type)) {
+    revng_assert(!Array->IsConst(),
+                 "Const arrays are not supported by this serializer.");
```

Comment:
Are they legal at all in the model?

Reply:
Nope, but I still felt like keeping the assert just to be extra safe.